tap_quickbooks/client.py

```diff
@@ -24,6 +23,94 @@ class Quickbooks5XXException(Exception):
 class Quickbooks4XXException(Exception):
     pass
 
+class QuickbooksAuthenticationError(Quickbooks4XXException):
+    pass
+
+class QuickbooksBadRequestError(Quickbooks4XXException):
+    pass
+
+class QuickbooksForbiddenError(Quickbooks4XXException):
+    pass
+
+class QuickbooksNotFoundError(Quickbooks4XXException):
+    pass
+
+class QuickbooksInternalServerError(Quickbooks5XXException):
+    pass
+
+class QuickbooksServiceUnavailableError(Quickbooks5XXException):
+    pass
+
+# Documentation: https://developer.intuit.com/app/developer/qbo/docs/develop/troubleshooting/error-codes
+ERROR_CODE_EXCEPTION_MAPPING = {
+    400: {
+        "exception": QuickbooksBadRequestError,
+        "message": "The request can't be fulfilled due to bad syntax."
+    },
+    401: {
+        "exception": QuickbooksAuthenticationError,
+        "message": "Authentication or authorization failed. Usually, this means the token in use won't work for API calls since it's either expired or revoked."
+    },
+    403: {
+        "exception": QuickbooksForbiddenError,
+        "message": "The URL exists, but it's restricted. External developers can't use or consume resources from this URL."
+    },
+    404: {
+        "exception": QuickbooksNotFoundError,
+        "message": "Couldn't find the requested resource or URL, or it doesn't exist."
+    },
+    500: {
+        "exception": QuickbooksInternalServerError,
+        "message": "A server error occurred while processing the request."
+    },
+    503: {
+        "exception": QuickbooksServiceUnavailableError,
+        "message": "The service is temporarily unavailable at the Server side.",
+    }
+}
+
+def get_exception_for_error_code(error_code):
+    """Function to retrieve exceptions based on error_code"""
+
+    exception = ERROR_CODE_EXCEPTION_MAPPING.get(error_code, {}).get('exception')
+    # If the error code is not from the listed error codes then return Quickbooks5XXException, Quickbooks4XXException or QuickbooksError respectively
+    if not exception:
+        if error_code >= 500:
+            return Quickbooks5XXException
+        if error_code >= 400:
+            return Quickbooks4XXException
+        return QuickbooksError
```

Comment:
Instead of writing 2 ifs, you can write one if and elif each
    if not exception:
        if error_code >= 500:
            return Quickbooks5XXException
        elif error_code >= 400:
            return Quickbooks4XXException
        return QuickbooksError